src/csync/CMakeLists.txt

```diff
@@ -84,6 +79,9 @@ if(ZLIB_FOUND)
   target_link_libraries(nextcloud_csync PUBLIC ZLIB::ZLIB)
 endif(ZLIB_FOUND)
 
+if(SQLite3_FOUND)
+  target_link_libraries(nextcloud_csync PRIVATE SQLite::SQLite3)
+endif(SQLite3_FOUND)
```

Comment:
would a build work without linking to `SQLite::SQLite3` ?
if no then remove the `if` and lets get a proper error